platforms/android/demo/src/main/java/com/mapzen/tangram/android/MainActivity.java

```diff
@@ -31,36 +40,130 @@
 
 import java.io.File;
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 
-public class MainActivity extends Activity implements OnMapReadyCallback, TapResponder,
-        DoubleTapResponder, LongPressResponder, FeaturePickListener, LabelPickListener, MarkerPickListener, SceneUpdateErrorListener {
+public class MainActivity extends AppCompatActivity implements OnMapReadyCallback, TapResponder,
+        DoubleTapResponder, LongPressResponder, FeaturePickListener, LabelPickListener, MarkerPickListener, SceneUpdateErrorListener,
+        AdapterView.OnItemSelectedListener, AdapterView.OnItemClickListener {
+
+    private final String apiKey = "vector-tiles-tyHL4AY";
+    private ArrayList<SceneUpdate> sceneUpdates = new ArrayList<>();
+
+    /* Override ArrayAdaptor to include any text */
+    private class CustomArrayAdapter extends ArrayAdapter {
```

Comment:
Make sense, now done in a separate class `DemoSceneManager`. The most apt name I could come for `CustomArrayAdapter` is `SceneAutoCompleteArrayAdapter`.